maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -847,12 +847,31 @@ private DefaultScanResult scanDependencies()
         {
             try
             {
+                DefaultScanResult scanResult = new DefaultScanResult( Collections.EMPTY_LIST );
+                
+                List<String> dependenciesToScan = new ArrayList();
+                Collections.addAll( dependenciesToScan, getDependenciesToScan() );
+                TestListResolver includedAndExcludedTests = getIncludedAndExcludedTests();
+
+                for ( MavenProject mavenProject : session.getSortedProjects() )
+                {
+                    String groupArtifactId = mavenProject.getGroupId() + ":" + mavenProject.getArtifactId();
+                    if ( dependenciesToScan.removeAll( Collections.singleton( groupArtifactId ) ) )
+                    {
+                        File outputDirectoryFile = new File( mavenProject.getBuild().getOutputDirectory() );
+                        DirectoryScanner scanner =
+                                new DirectoryScanner( outputDirectoryFile, includedAndExcludedTests );
+                        scanResult = scanResult.append( scanner.scan() );
+                    }
+                }
+
                 // @TODO noinspection unchecked, check MavenProject 3.x for Generics in surefire:3.0
                 @SuppressWarnings( "unchecked" )
-                List<File> dependenciesToScan =
-                    DependencyScanner.filter( project.getTestArtifacts(), Arrays.asList( getDependenciesToScan() ) );
-                DependencyScanner scanner = new DependencyScanner( dependenciesToScan, getIncludedAndExcludedTests() );
-                return scanner.scan();
+                List<File> dependenciesToScanFile =
```

Comment:
Pls align to single line after using static import:
`import static java.lang.Thread.currentThread;
import static org.apache.maven.plugin.surefire.util.DependencyScanner.filter;
import static java.util.Collections.singletonMap;`